docs/design/authentication_policy.md

```diff
@@ -0,0 +1,40 @@
+# Envoy Gateway Authentication Policy API
+
+## Overview
+
+This authen policy is to declare the authentication mechanisms, to be enforce on connection and request going though Envoy Gateway. This includes the credential (X.509, JWT, etc), parameters (cipher suites, key algorithms)
+The policy is similar to [OpenAPI 3.1 security objects](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#securitySchemeObject) without the API key part, and should be easily translatable from it with some additions.
+
+## Authentication mechanisms
+This policy should support the following authentication mechanisms:
+- JWT Bearer Token
+- mutualTLS (client certificate)
+- OAuth2
+- OIDC
+- External authentication
+
+In the first phase, Envoy Gateway will implement JWT Bearer Token authentication.
+
+In general those policy translates into Envoy HTTP filters at HTTP connection manager level, and route specific settings will be applied for each route. These APIs are expressed in a Policy CRD and attached to Gateway API resources with [Policy Attachement](https://gateway-api.sigs.k8s.io/references/policy-attachment/).
+
+## JWT Bearer Token
+
+A JWT Bearer Token authentication policy will look like the following:
+
+```
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: Authentication
+metadata:
+  name: productpage
+spec:
+  type: jwt
```

Comment:
are we going to expose `from_headers` ? I'd vote for not now, and prefer to use the defaults in Envoy 
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/jwt_authn/v3/config.proto#extensions-filters-http-jwt-authn-v3-jwtprovider

Reply:
+1

Reply:
#733 does not specify `from_headers`. As you mention, this can be introduced in the future.